src/erlcloud_aws.erl

```diff
@@ -769,34 +769,43 @@ profiles_resolve( Name, Profiles, Role, Options ) ->
 
 profiles_recurse( Keys, Profiles, Role, Options ) ->
     case profiles_credentials( Keys ) of
-        {ok, Id, Secret} ->
-            profiles_assume( Id, Secret, Role, Options );
+        {ok, Credential} ->
+            profiles_assume( Credential, Role, Options );
         {cont, ProfileName, BinRole} ->
             profiles_resolve( ProfileName, Profiles, BinRole, Options )
     end.
 
 profiles_credentials( Keys ) ->
-    Names = [aws_access_key_id, aws_secret_access_key, source_profile],
+    Names = [aws_access_key_id, aws_secret_access_key, aws_security_token, source_profile],
     BinRole = proplists:get_value( role_arn, Keys ),
     case [proplists:get_value( K, Keys ) || K <- Names] of
-        [Id, Secret, undefined] when Id =/= undefined, Secret =/= undefined ->
-            {ok, binary_to_list(Id), binary_to_list(Secret)};
-        [undefined, undefined, BinProfile] when BinProfile =/= undefined ->
+        [Id, Secret, undefined, undefined] when Id =/= undefined, Secret =/= undefined ->
+            {ok, {binary_to_list(Id), binary_to_list(Secret)}};
+        [Id, Secret, Token, undefined] when Id =/= undefined, Secret =/= undefined, Token =/= undefined ->
+            {ok, {binary_to_list(Id), binary_to_list(Secret), binary_to_list(Token)}};
+        [undefined, undefined, _, BinProfile] when BinProfile =/= undefined ->
```

Comment:
i would prefer to be as close as possible to AWS CLI, where together with source profile user sets only the role.
 so `[undefined, undefined, undefined, BinProfile]`